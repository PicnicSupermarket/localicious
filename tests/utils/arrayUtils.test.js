const { groupByKey } = require("../../src/utils/arrayUtils");

test("grouping an array", () => {
  let list = [
    {
      language: "nl",
      translation: "Een vertaling",
    },
    {
      language: "en",
      translation: "A translation",
    },
  ];
  let res = groupByKey(list, (item) => item.language);
  expect(res).toMatchSnapshot();
});
