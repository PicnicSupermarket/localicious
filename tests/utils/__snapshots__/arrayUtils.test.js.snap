// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grouping an array 1`] = `
Object {
  "en": Array [
    Object {
      "language": "en",
      "translation": "A translation",
    },
  ],
  "nl": Array [
    Object {
      "language": "nl",
      "translation": "Een vertaling",
    },
  ],
}
`;
